digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/service/feedback/FeedbackOperations.scala

```diff
@@ -0,0 +1,171 @@
+package fi.liikennevirasto.digiroad2.service.feedback
+
+import java.util.Properties
+
+import fi.liikennevirasto.digiroad2.asset._
+import fi.liikennevirasto.digiroad2.{Email, EmailOperations}
+import fi.liikennevirasto.digiroad2.dao.feedback.FeedbackDao
+import fi.liikennevirasto.digiroad2.oracle.OracleDatabase
+import fi.liikennevirasto.digiroad2.util.EmailAuthPropertyReader
+import javax.mail.MessagingException
+import org.joda.time.DateTime
+
+case class FeedbackInfo(id: Long, receiver: Option[String], createdBy: Option[String], createdAt: Option[DateTime], body: Option[String],
+                        subject: Option[String], status: Boolean, statusDate: Option[DateTime])
+
+case class FeedbackApplicationBody(feedbackType: Option[String], headline: Option[String], freeText: Option[String], name: Option[String], email: Option[String], phoneNumber: Option[String])
+case class FeedbackDataBody(linkId: Option[Seq[Long]], assetId: Option[Seq[Long]], assetName: Option[String], feedbackDataType: Option[String], freeText: Option[String], name: Option[String], email: Option[String], phoneNumber: Option[String], typeId: Option[Int])
+
+trait FeedbackOperations {
+
+  def withDynTransaction[T](f: => T): T = OracleDatabase.withDynTransaction(f)
+  def withDynSession[T](f: => T): T = OracleDatabase.withDynSession(f)
+  def dao: FeedbackDao
+  def emailOperations: EmailOperations
+  def to: String
+  def from: String
+  def subject: String
+  def body: String
+  private def smtpHost = getProperty("digiroad2.feedbackSmtpHost")
+  private def smtpPort= getProperty("digiroad2.feedbackSmtpPort")
+  type FeedbackBody
+
+  lazy val dr2properties: Properties = {
+    val props = new Properties()
+    props.load(getClass.getResourceAsStream("/digiroad2.properties"))
+    props
+  }
+
+  protected def getProperty(name: String) = {
+    val property = dr2properties.getProperty(name)
+    if(property != null)
+      property
+    else
+      throw new RuntimeException(s"cannot find property $name")
+  }
+
+
+  def stringifyBody(username: String, body: FeedbackBody) : String
+
+  def insertFeedback(username: String, body: FeedbackBody): Long = {
+    val message = stringifyBody(username, body)
+    withDynSession {
+      dao.insertFeedback(to, username, message, subject, status = false)
```

Comment:
the `to ` can be removed from the feedback table,since we are going to get it from the properties file